Userland/Libraries/LibTest/JavaScriptTestRunnerMain.cpp

```diff
@@ -32,7 +32,17 @@ static StringView g_program_name { "test-js"sv };
 static void handle_sigabrt(int)
 {
     dbgln("{}: SIGABRT received, cleaning up.", g_program_name);
-    cleanup_and_exit();
+    cleanup();
+    struct sigaction act;
+    memset(&act, 0, sizeof(act));
+    act.sa_flags = SA_NOCLDWAIT;
+    act.sa_handler = SIG_DFL;
+    int rc = sigaction(SIGABRT, &act, nullptr);
```

Comment:
Hrm I don't really like re-assigning a signal handler while within that same signal handler. I think Dr Posix prescribes using the SA_RESETHANDLER flag to the original sigaction to do this? Don't think that's implemented yet though. Struggling to find a better way to do this though, we really don't want the test to keep going after the assertion failed. Sets my "wait I read something about this on SEI CERT coding standards" alarms off.

https://wiki.sei.cmu.edu/confluence/display/c/SIG01-C.+Understand+implementation-specific+details+regarding+signal+handler+persistence

https://wiki.sei.cmu.edu/confluence/display/c/SIG34-C.+Do+not+call+signal%28%29+from+within+interruptible+signal+handlers

I *think* the way it's handled here is correct though...

Reply:
the test will not keep going, the handler will just `abort()` after resetting the handler, which will do whatever the default handler is supposed to do - i.e. kill the program.
Your links are about making a persistent signal handler, the one thing this chunk of code tries to *avoid* lol

I don't think I want to implement `SA_RESETHANDLER` for this, and AFAIK resetting the handler like this is fairly common :shrug: